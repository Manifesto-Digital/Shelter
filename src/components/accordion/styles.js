import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
