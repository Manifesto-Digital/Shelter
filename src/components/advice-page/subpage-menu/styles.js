import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
