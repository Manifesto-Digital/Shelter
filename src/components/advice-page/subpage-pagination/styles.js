import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
