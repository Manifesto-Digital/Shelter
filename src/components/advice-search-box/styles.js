import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
