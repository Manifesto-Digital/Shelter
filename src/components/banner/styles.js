import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
