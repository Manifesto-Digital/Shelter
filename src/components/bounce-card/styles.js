import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
