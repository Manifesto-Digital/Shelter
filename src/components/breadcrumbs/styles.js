import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
