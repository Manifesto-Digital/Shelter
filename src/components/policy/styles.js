import styled from 'styled-components';
import DownloadCTA from '../download-cta';

export const PolicyCard = styled.div`
  background-color: ${({ theme }) => theme.palette.grey10};
  padding: ${({ theme }) => theme.spacing.standard};
  box-shadow: ${({ theme }) => theme.boxshadow.small};
  margin-bottom: ${({ theme }) => theme.spacing.standard};
  display: flex;
  flex-wrap: wrap;
`;

export const Detail = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.small};
  margin-right: ${({ theme }) => theme.spacing.medium};

  &:last-child {
    margin-left: 0;
  }
`;

export const DocumentDownload = styled(DownloadCTA)`
  min-width: 75%;
`;
