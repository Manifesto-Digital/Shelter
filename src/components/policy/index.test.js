import React from 'react';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../__tests__/helpers/index';
import {
  createFactory,
  createChildContentfulRichText,
  createFile,
} from '../../utils/test-factories';
import Policy from './index';
import { Detail } from './styles';
import { dateAsString } from '../../utils/dates';

export const createPolicy = createFactory({
  policyName: 'My very cool policy',
  author: 'Owen Jones',
  publishDate: '2019-02-19',
  displayDate: 'February 2019',
  summary: createChildContentfulRichText(),
  media: [createFile()],
});

it('renders correctly', () => {
  const mockData = createPolicy();

  snapshotComponent(<Policy data={mockData} />);
});

it('displays the default date if no display date is provided', () => {
  const mockData = createPolicy({
    publishDate: '2020-01-01',
    displayDate: '',
  });

  const wrapper = mountWithTheme(<Policy data={mockData} />);

  expect(
    wrapper
      .find(Detail)
      .at(1)
      .text()
  ).toContain(dateAsString(mockData.publishDate, 'DD/MM/YYYY'));
});
