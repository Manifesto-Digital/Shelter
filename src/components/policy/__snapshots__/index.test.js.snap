// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px 20px;
  background-color: #8c8c8c;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  font-size: 0.65em;
  margin-bottom: 5px;
}

.c4 {
  margin-bottom: 5px;
}

.c6 {
  display: block;
  fill: #fff;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.c0 {
  background-color: #e5e5e5;
  padding: 20px;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  margin-bottom: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  margin-bottom: 10px;
  margin-right: 30px;
}

.c1:last-child {
  margin-left: 0;
}

.c2 {
  min-width: 75%;
}

@media (min-width:40.0625em) {
  .c6 {
    width: 25px;
    height: 25px;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <strong>
        By:
      </strong>
       
      Owen Jones
    </div>
    <div
      className="c1"
    >
      <strong>
        Published:
      </strong>
       
      February 2019
    </div>
    <h4>
      My very cool policy
    </h4>
    <a
      className="c2 c3"
      download={true}
      href="//assets.ctfassets.net/6sxvmndnpn0s/6gDmPHZ04gCMMS4sOa0IEk/7c712c5e8e383bbafd61f2d9d570de4f/component_tree-pdf.pdf"
    >
      <div>
        <p
          className="c4"
        >
          Mock file title
        </p>
        <p
          className="c5"
        >
          78.02 KB
        </p>
      </div>
      <span
        className="isvg loading c6"
      />
    </a>
  </div>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Our advice and support services across the UK give peopleÂ <b>one-to-one</p>",
      }
    }
  />,
]
`;
